pkg/rpc/nodedialer/nodedialer_test.go

```diff
@@ -163,9 +163,12 @@ func TestConnHealth(t *testing.T) {
 	// Closing the remote connection should fail ConnHealth.
 	require.NoError(t, ln.popConn().Close())
 	hbDecommission.Store(true)
-	require.Eventually(t, func() bool {
-		return nd.ConnHealth(staticNodeID, rpc.DefaultClass) != nil
-	}, time.Second, 10*time.Millisecond)
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		assert.NotNil(c, nd.ConnHealth(staticNodeID, rpc.DefaultClass),
```

Comment:
Just double checking since I haven't seen this pattern before: will this `assert` not cause the test to fail if the check fails at least once?

Reply:
Or we're preventing that by redirecting it to this `CollectT` thing?

Reply:
This one is a different kind of assert than what we usually use, from the same pkg author and used in the `require` lib.